bundles/org.eclipse.swt/Eclipse SWT/gtk/org/eclipse/swt/widgets/Shell.java

```diff
@@ -479,20 +479,42 @@ void addToolTip (ToolTip toolTip) {
 	toolTips = newToolTips;
 }
 
-void adjustTrim () {
+/**
+ * Resize the shell based on the trim width and height. Trim width and height
+ * adjustments are read from 'user/.swt.trims.prefs' for each trim style. If
+ * such a file does not exist then it uses default values initialized.
+ *
+ * Note that we are trying to resize the shell which is already open. Shell
+ * already has width and height, Which is either set by setSize()/setBounds() or
+ * size is calculated based on layout and content of the shell. Without setting
+ * bounds/layout shell won't display content which is created on it.
+ *
+ * @param curWidth  Current width of the shell that is already open.
+ * @param curHeight Current height of the shell that is alreday open.
+ */
+void adjustTrim (int curWidth, int curHeight) {
 	if (display.ignoreTrim) return;
 	GtkAllocation allocation = new GtkAllocation ();
 	GTK.gtk_widget_get_allocation (shellHandle, allocation);
 	int width = allocation.width;
 	int height = allocation.height;
-	GdkRectangle rect = new GdkRectangle ();
 
+	//Ask Gtk for actual window frame extents
+	GdkRectangle rect = new GdkRectangle ();
 	if (!GTK.GTK4) {
 		long window = gtk_widget_get_window (shellHandle);
 		GDK.gdk_window_get_frame_extents (window, rect);
 	}
-	int trimWidth = Math.max (0, rect.width - width);
-	int trimHeight = Math.max (0, rect.height - height);
+
+	// Gtk returns default window dimensions as {0,0,1,1} if the window is not
+	// ready. If Gtk returns window dimensions we can use it. See 1161
```

Comment:
`If Gtk returns window dimensions we can use it.` 
->  
`If GTK returns window dimensions we can use it, if not, we will use "requested" shell size as returned by gtk_window_resize() in setVisible()`

`See 1161` -> `See https://github.com/eclipse-platform/eclipse.platform.swt/issues/1161`